### PR TITLE
Feature/moz 323 attach campaigns to events

### PR DESCRIPTION
**Pull request by dnaluz**
This PR attaches campaigns and activities to events.  It adds campaign and activity filters to the event listing page, campaign and activity links to the event pages as well. You can compare the pages to the figma files for reference.

To create events and activities you'll need to install the new ACF fields for each content type.  You can find the json file to install the new fields in the theme folder under the acf directory and use the ACF tools to import them.

### CODE REVIEW

**Comment by kycapstick** [comment]
This is all looking really great. Feedback below:

ACTIVITIES

- If there are no events attached to the activity it's throwing errors. 
- Not sure if this is something I need to set up, but I'm getting the following error on single-activity.php lines 110, 157 and 161 and  page-activities.php line does 58:

"Parameter must be an array or an object that implements Countable " 

- Tied to the above (the $tags variable), getting the following error in single-activity.php on line 113: 

"Invalid argument supplied for foreach()"

- Time commitment isn't showing up on the Activities card. 

CAMPAIGNS

- Tested publishing with just a title and getting the following error on line 61 of single-campaign.php (no blocks were added):

"Invalid argument supplied for foreach()"

- All the text added in the two-up and three-up blocks is centered, but should be left aligned. 
- Line height in unordered and ordered lists is really tight. 
- If copy in one column is shorter than copy in any of the other columns, the copy ends up centered. Not sure if this is the desired design. 
- three-up block is rendering wider than any of the other blocks. It looks like all blocks should be the same width. 
- Column 1 Cta URL and Column 2 CTA Url are both set to text fields, but should be URL fields. 
- If copy on image/text block is long, image stays fixed at top of container, but I believe it should be centred vertically (note says it should be centred horizontally, but I think they mean vertically)
- Event cards are in events block are missing Campaign/Activity association.
- Date on event was set to Feb 8, and is showing in one line, when should have month on one line and date on second line. 
- If event is associated with campaign and shows up in Events Block, it has a grey bar across the bottom 

Screen Shot 2020-02-05 at 2 38 27 PM

- Images in Imagery Block are uploading at different heights. Not sure if we want to standardize this so they render at the same height? 
- CTA Link in Outro block is text input but should be a URL

- I was initially having issues because I hadn't set an active campaign on the Campaigns page. If one isn't set, it still shows the card at the top with the "Get Involved" button. Should we be proving support for if there is no currently active campaign? 
- If a campaign is technically active (today's date is between its start and end dates), but not set as the active campaign in the Campaigns page fields it doesn't show up. Is there a use case where there will be more than one active campaign at a time? 
- Also, if a campaign has technically closed but still left as the "Active Campaign" in the Campaigns page fields, it shows up in both places. 
- Similarly, if there is no "Incoming campaign" set, it seems to automatically populate the next upcoming campaign. But if a past campaign is left in this field on the Campaigns page, it still shows up as incoming.

**Comment by dnaluz** [comment]
@kycapstick 

> three-up block is rendering wider than any of the other blocks. It looks like all blocks should be the same width.

According to the figma files 
Screen Shot 2020-02-05 at 3 28 35 PM

**Comment by dnaluz** [comment]
@kycapstick 

> All the text added in the two-up and three-up blocks is centred, but should be left aligned.

Screen Shot 2020-02-05 at 3 32 14 PM

**Comment by dnaluz** [comment]
Added array checks and a check if there is a current campaign to hide the current campaign card

**Comment by dnaluz** [comment]
@kycapstick 

> If a campaign is technically active (today's date is between its start and end dates), but not set as the active campaign in the Campaigns page fields it doesn't show up. Is there a use case where there will be more than one active campaign at a time?

To be active the status can't be closed and must fall between the dates.   This is a fail safe if there is none set on the campaigns page it takes the first one that falls within these parameters if not now it hides the current campaign card.  I recently added a check to hide the current campaign.

**Comment by dnaluz** [comment]
@kycapstick 

> Date on event was set to Feb 8, and is showing in one line, when should have month on one line and date on second line.

We are ok with the date flowing with the width of the card.

**Comment by dnaluz** [comment]
@kycapstick 

> I was initially having issues because I hadn't set an active campaign on the Campaigns page. If one isn't set, it still shows the card at the top with the "Get Involved" button. Should we be proving support for if there is no currently active campaign?

Added check to hide the card

**Comment by dnaluz** [comment]
@kycapstick 

> Also, if a campaign has technically closed but still left as the "Active Campaign" in the Campaigns page fields, it shows up in both places.

We should ask Mozilla what they think of this.  I will bring it up during our meeting tomorrow.

**Comment by dnaluz** [comment]
@kycapstick 

> Tested publishing with just a title and getting the following error on line 61 of single-campaign.php (no blocks were added):

Added array check

**Comment by dnaluz** [comment]
@kycapstick 

> Column 1 Cta URL and Column 2 CTA Url are both set to text fields, but should be URL fields.

> CTA Link in Outro block is text input but should be a URL

Added updated json file in the acf folder.  New date.

**Comment by dnaluz** [comment]
@kycapstick 

> Line height in unordered and ordered lists is really tight.

Which block is this for?

**Comment by dnaluz** [comment]
@kycapstick 

>If event is associated with campaign and shows up in Events Block, it has a grey bar across the bottom

This is an issue with the event container.  Added height 100% to fix this issue.

**Comment by dnaluz** [comment]
@kycapstick 

> Event cards are in events block are missing Campaign/Activity association.

Added initiative to events block

**Comment by kycapstick** [comment]
> @kycapstick
> 
> > Line height in unordered and ordered lists is really tight.
> 
> Which block is this for?

This is showing up in the 2-up and 3-up columns. We might need to follow up with design how they'd like to handle them, since they also look a bit weird centered. 

Screen Shot 2020-02-06 at 10 41 32 AM

**Comment by dnaluz** [comment]
Updated how imagery block is displaying the image and added the background options for images that don't scale / fit properly

**Comment by dnaluz** [comment]
@kycapstick 

I just added line height for content that shows up in the container

Screen Shot 2020-02-06 at 11 01 15 AM

**Comment by dnaluz** [comment]
@kycapstick I think I've address all the issues.  Let me know

**Comment by kycapstick** [comment]
This is all looking great! Left a couple of comments in places where it's throwing errors if no tags are applied to a Campaign. The only other things I'm noticing are things I think we should clarify with design.

- For the 3-up column, if the text isn't the same length it centers them rather than keeping them at the top. 
Screen Shot 2020-02-06 at 2 20 07 PM

- Did we ever get confirmation on whether the image should be centered vertically in the image/text block, I saw the question posted in the channel. 

Screen Shot 2020-02-06 at 2 21 24 PM

**Comment by dnaluz** [comment]
> Did we ever get confirmation on whether the image should be centered vertically in the image/text block, I saw the question posted in the channel.

Spoke with Dondy, top aligning the image is what he decided on for this block

**Comment by dnaluz** [comment]
@kycapstick 

> For the 3-up column, if the text isn't the same length it centers them rather than keeping them at the top.

Changed this to top aligning the content for tablet and up view ports

**Comment by kycapstick** [comment]
This is looking great. Two last small things. 

- Still getting the sizeof error on line 47 of single-campaigns
- The fix to top align the content needs to also be applied to the two-up columns

**Comment by dnaluz** [comment]
@kycapstick last two issues have been fixed.  Let me know if there is anything else!

File: page-activities.php [hunk 1]
```diff
@@ -55,7 +55,7 @@
                             $tags = get_the_tags($activity->ID);
                         ?>
                         <div class="activities__tag-container">
-                            <?php if(sizeof($tags) > 0): ?>
+                            <?php if(is_array($tags) && sizeof($tags) > 0): ?>
```
Comment (dnaluz) [inline page-activities.php]: @kycapstick added is array check here, you shouldn't receive the error anymore

Comment (kycapstick) [reply page-activities.php]: This looks great. The last thing is that same fix needs to be added to the single-activities page.

Comment (dnaluz) [reply page-activities.php]: Array check was in the incorrect order on the single-activities page.  Should be good now

Comment (kycapstick) [reply page-activities.php]: Looks great! Good to merge!

**Pull request merged**